### PR TITLE
feat: Add icon to searchbar

### PR DESCRIPTION
**Pull request by danschewy**
CodeSandbox logoCodeSandbox logo&nbsp; Open in CodeSandbox <a href="https://codesandbox.io/p/github/danschewy/lenster/feat/search-icon?workspace=%7B%22gitSidebarPanel%22:%22PR%22,%22sidebarPanel%22:%22GIT%22%7D">Web Editor</a> | <a href="https://codesandbox.io/p/vscode?owner=danschewy&repo=lenster&branch=feat/search-icon">VS Code</a>

Add a left-icon prop to input, add SearchIcon to search input in mobile view

unfocused
Screen Shot 2022-07-29 at 3 56 57 AM

focused
Screen Shot 2022-07-29 at 3 57 08 AM

disabled
Screen Shot 2022-07-29 at 4 24 04 AM

unfocused error
Screen Shot 2022-07-29 at 4 47 28 AM

focused error
Screen Shot 2022-07-29 at 4 47 38 AM

### CODE REVIEW

**Comment by danschewy** [comment]
Whoops did not account for disabled input...

**Comment by danschewy** [comment]
Not sure how to handle icon and ring color in error state, if you like this approach please advice

**Comment by bigint** [comment]
Screenshot 2022-07-29 at 4 59 19 PM

@danschewy I'm not seeing the search icon 🤔

**Comment by danschewy** [comment]
> Screenshot 2022-07-29 at 4 59 19 PM
> 
> @danschewy I'm not seeing the search icon 🤔

I am passing sm:hidden on SearchIcon in Search.tsx :)

**Comment by bigint** [comment]
@danschewy I think we need icon in both desktop and mobile 🙇

**Comment by bigint** [comment]
Screenshot 2022-07-30 at 11 25 00 AM

@danschewy I'm seeing it's defaulted to red border 🤔

File: src/components/Shared/Navbar/Search.tsx [hunk 1]
```diff
@@ -75,6 +76,8 @@ const Search: FC<Props> = ({ hideDropdown = false }) => {
             className="py-2 px-3 text-sm"
             placeholder="Search..."
             value={searchText}
+            error
+            leftIcon={<SearchIcon />}
```
Comment (bigint) [inline src/components/Shared/Navbar/Search.tsx:80]:
```suggestion
            value={searchText}
            leftIcon={<SearchIcon />}
```

**Comment by shanpriyan** [comment]
@bigint this PR resolves #68

**Comment by danschewy** [comment]
> Screenshot 2022-07-30 at 11 25 00 AM
> 
> @danschewy I'm seeing it's defaulted to red border 🤔

I must have checked in the “error” when testing the styling 😔

Thank you for making the fix and merging! 🙏

Will look at other issues and see how I can help in my free time 😊

File: src/components/UI/Input.tsx [hunk 2]
```diff
@@ -32,23 +42,43 @@ export const Input = forwardRef<HTMLInputElement, Props>(function Input(
...
+            ref={ref}
+            {...props}
+          />
+          <span
+            className={clsx(
+              { '!text-red-500 [&>*]:peer-focus:!text-red-500': error },
+              'order-first text-zinc-500 [&>*]:peer-focus:text-brand-500 [&>*]:h-5'
+            )}
+          >
+            {leftIcon}
```
Comment (shanpriyan) [inline src/components/UI/Input.tsx:79]: Shall we rename the prop `leftIcon` --> `rightIcon`, since it is being used on the right side of the searchbar

**Comment by shanpriyan** [comment]
@danschewy on `focused && error` state it is showing two colors combined in border. Let's remove the `ring` for `focused && error` state.
image

**Comment by bigint** [comment]
@shanpriyan @danschewy added some more stuffs in this PR https://github.com/lensterxyz/lenster/pull/177

**Comment by danschewy** [comment]
> @danschewy on `focused && error` state it is showing two colors combined in border. Let's remove the `ring` for `focused && error` state. image

https://github.com/lensterxyz/lenster/pull/180

**Pull request merged**